File: coding-patterns/00-warmup/007.cpp
// Number of Good Pairs (Easy)

#include <iostream>
#include <map>
#include <iostream>
#include <vector>
#include <unordered_map>

class Solution {
  public:
    int numGoodPairs(std::vector<int>& nums) {
      int pairCount = 0;
      std::unordered_map<int, int> hmap;
      for (auto num : nums) {
        hmap[num];
        hmap[num]++;
        pairCount += hmap[num] - 1;
      }
      return pairCount;
    }
};

File: coding-patterns/00-warmup/007.py
# Number of Good Pairs (Easy)

class Solution:
  def numGoodPairs(self, nums):
    pairCount = 0
    map = {}
    for num in nums:
      map[num] = map.get(num , 0) + 1 
      pairCount += map[num] - 1
    return pairCount
